rosidl_typesupport_introspection_cpp/resource/msg_TypeSupport_Introspection.cpp.template

```diff
@@ -19,6 +19,24 @@
 #include "rosidl_generator_cpp/MessageTypeSupport.h"
 #include "rosidl_typesupport_introspection_cpp/FieldTypes.h"
 #include "rosidl_typesupport_introspection_cpp/MessageIntrospection.h"
+#include "rosidl_typesupport_introspection_cpp/impl/visibility_control.h"
+
+@{
+import os
+
+IS_WINDOWS = os.name == 'nt'
+
+if not IS_WINDOWS:
+    print('#define offsetof __builtin_offsetof')
```

Comment:
There's a warning on Linux because `offsetof` is already defined:

```
/home/esteve/Projects/ros2/ros2_ws/build/builtin_msgs/rosidl_typesupport_introspection_cpp/builtin_msgs/Duration_TypeSupport_Introspection.cpp:11:0: warning: "offsetof" redefined [enabled by default]
 #define offsetof __builtin_offsetof
 ^
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from /home/esteve/Projects/ros2/ros2_ws/install/include/rosidl_typesupport_introspection_cpp/MessageIntrospection.h:4,
                 from /home/esteve/Projects/ros2/ros2_ws/build/builtin_msgs/rosidl_typesupport_introspection_cpp/builtin_msgs/Duration_TypeSupport_Introspection.cpp:8:
/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:413:0: note: this is the location of the previous definition
 #define offsetof(TYPE, MEMBER) __builtin_offsetof (TYPE, MEMBER)
```